library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -591,23 +591,33 @@ protected void replay(long windowId)
                 pendingFiles.remove(recoveryEntry.file);
               }
               inputStream = retryFailedFile(new FailedFile(recoveryEntry.file, recoveryEntry.startOffset));
+
```

Comment:
There is similar code to skip tuples in emitTuples(), but I believe it will never be executed since there is a check at the top of that function that simply returns if recovery is under way. Additionally, I think this check will _never_ be true:
{code}
if (inputStream == null) {
      try {
        if (currentFile != null && offset > 0) {
        ....
{code}

This logic should be carefully examined and all dead code removed from emitTuples().